### ISSUE TITLE
Client crash when setting forge "global config" option to true.

### ISSUE THREAD

**Issue by katubug** [KnightMiner/Inspirations#144]
Minecraft version: 1.12.2
Forge version: forge-1.12.2-14.23.5.2838-universal
Mod versions: Inspirations 0.2.8

Link to crash log: https://paste.dimdev.org/corezumewo.mccrash

Everything was working fine in modpack version 1.6.0, the modlist for which is here: http://enthusiasticraft.cf/modpack-update-1-6-0/#modlist

Currently running some tests in a controlled environment with just the most recently updated mods. Gonna see if I can isolate which mod in specific is causing the issue.

**Comment by katubug** [comment]
Scroll to bottom for tl;dr. Sorry for the lengthy troubleshooting, I like to make sure I test rigorously.

- Loading up the game with JUST the newly updated mods plus their dependencies: doesn't crash, oddly.
- Loading up the game with JUST the newly updated mods plus their dependencies, *and the config folder from the main modpack*: does crash. Here is the log: https://pastebin.com/AeGZL5Y7
- Loading up the main modpack with the Inspirations config deleted (but I forgot InspirationsModules.cfg): crashes. Floating decimal error again.
- Loading up the main modpack with both inspirations.cfg and inspirationsmodules.cfg deleted: crashes. Same error again.

- Starting with just newly updated mods, default configs.
-- Adding in inspirations.cfg and inspirationsmodules.cfg: no crash.
-- Adding in folders for jei and forgeendertech: no crash.
-- Adding in folder for thebetweenlands, and betteranimalsplus.cfg: no crash.
-- Adding in Building Gadgets.cfg, cyclicmagic.cfg, and cyclic_ores.cfg: no crash.
-- Adding in forge.cfg and forgeChunkLoading.cfg: crash! Same error.
-- Removed forge.cfg and forgeChunkLoading.cfg, added in the rest of the configs: no crash.
-- Adding back forge.cfg: crash. Same error.
-- Just to be safe, deleting forge.cfg and using the modpack version of forgeChunkLoading.cfg: no crash.

Okay, so evidently the issue is with forge.cfg - however, I didn't update forge recently, so I'll have to see what the difference is between the two configs.

Disagreeing values:
```
# Log cascading chunk generation issues during terrain population.
B:logCascadingWorldGeneration=false
```
	
The old version has this set to false, default is true. However, I think I've had this setting for a while, so I don't think it's the issue. Here's the other value:
	
```
# The time in ticks the server will wait when a dimension was queued to unload. This can be useful when rapidly loading and unloading dimensions, like e.g. throwing items through a nether portal a few time per second.
I:dimensionUnloadQueueDelay=20
B:enableGlobalConfig=true
```
	
The default for these is `I:dimensionUnloadQueueDelay=0` and `B:enableGlobalConfig=false`.

I'll test to see which is the issue.
- "B:logCascadingWorldGeneration=false" all else default: no crash.
- "I:dimensionUnloadQueueDelay=20": no crash.
- "B:enableGlobalConfig=true": crash, as expected.

**In conclusion, the issue is with setting forge's "global config" option to true.**

**Comment by KnightMiner** [comment]
Does this happen when enabling global config with a fresh inspirations config? The crash log says you have an invalid config version in the Inspirations file.

If it does still crash, please give me a copy of your inspirations.cfg file.

**Comment by katubug** [comment]
I deleted all configs except for forge.cfg, and it does still crash, before the inspirations.cfg can regenerate. However, it does generate a global.cfg with some Inspirations-related options: https://pastebin.com/qdDKHrie

If I change the "global config" option in forge.cfg to false, the inspirations.cfg does generate, and here it is: https://pastebin.com/vZBhz9a9

Edit: I don't know what purpose the global config serves, I think I enabled it because I assumed it was related to the dimension unload delay. I shall disable it to fix the issue on my end.

**Comment by KnightMiner** [comment]
Fixed in 0.2.9
